xwiki-platform-core/xwiki-platform-tag/xwiki-platform-tag-api/src/main/java/org/xwiki/tag/internal/livedata/TaggedDocumentLiveDataConfigurationProvider.java

```diff
@@ -0,0 +1,77 @@
+/*
+ * See the NOTICE file distributed with this work for additional
+ * information regarding copyright ownership.
+ *
+ * This is free software; you can redistribute it and/or modify it
+ * under the terms of the GNU Lesser General Public License as
+ * published by the Free Software Foundation; either version 2.1 of
+ * the License, or (at your option) any later version.
+ *
+ * This software is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public
+ * License along with this software; if not, write to the Free
+ * Software Foundation, Inc., 51 Franklin St, Fifth Floor, Boston, MA
+ * 02110-1301 USA, or see the FSF site: http://www.fsf.org.
+ */
+package org.xwiki.tag.internal.livedata;
+
+import java.util.Collections;
+
+import javax.inject.Named;
+import javax.inject.Provider;
+import javax.inject.Singleton;
+
+import org.xwiki.component.annotation.Component;
+import org.xwiki.livedata.LiveDataConfiguration;
+import org.xwiki.livedata.LiveDataMeta;
+import org.xwiki.livedata.LiveDataPropertyDescriptor;
+import org.xwiki.livedata.LiveDataQuery;
+
+import static org.xwiki.livedata.LiveDataPropertyDescriptor.DisplayerDescriptor;
+
+/**
+ * Live data configuration for the tag.
+ *
+ * @version $Id$
+ * @since 13.1RC1
+ */
+@Component
+@Named("taggedDocument")
+@Singleton
+public class TaggedDocumentLiveDataConfigurationProvider implements Provider<LiveDataConfiguration>
+{
+    @Override
+    public LiveDataConfiguration get()
+    {
+        LiveDataConfiguration liveDataConfiguration = new LiveDataConfiguration();
+        LiveDataQuery query = new LiveDataQuery();
+        LiveDataQuery.Source source = new LiveDataQuery.Source();
+        source.setId("tag");
+        query.setSource(source);
+        liveDataConfiguration.setQuery(query);
+
+        LiveDataPropertyDescriptor liveDataPropertyDescriptor = new LiveDataPropertyDescriptor();
+        liveDataPropertyDescriptor.setId("page");
+        DisplayerDescriptor linkDisplayer = new DisplayerDescriptor("link");
+        linkDisplayer.setParameter("propertyHref", "page_link");
+        liveDataPropertyDescriptor.setDisplayer(linkDisplayer);
+        liveDataPropertyDescriptor.setSortable(false);
+        liveDataPropertyDescriptor.setEditable(false);
+        liveDataPropertyDescriptor.setVisible(true);
+        liveDataPropertyDescriptor.setFilterable(false);
+        liveDataPropertyDescriptor.setName("Page");
+        liveDataPropertyDescriptor.setType("document");
+
+        LiveDataMeta meta = new LiveDataMeta();
+        liveDataConfiguration.setMeta(meta);
+        meta.setPropertyDescriptors(Collections.singletonList(
+            liveDataPropertyDescriptor
```

Comment:
Stupid question: why is this all done in Java? It seems a regression from before when we could have that in a wiki page and thus be editable to adjust to the user's needs.

Reply:
@vmassol this is the **default** configuration for the ``tag`` live data source, so you don't have to duplicate it whenever you want to display the tagged documents using the live data macro: you just need to specify the source and the list of properties. The fact that it is done in Java doesn't imply it cannot be done from a wiki page. You can overwrite this default configuration when calling the live data macro.

Now, the class name should be improved (``Default...``) and the source id should probably be "taggedPages" or "taggedDocuments" because it doesn't return tags but pages / documents. I mean, a live data entry is a page / document, not a tag.